library/Makefile

```diff
@@ -2,7 +2,7 @@
 # Also see "include/mbedtls/config.h"
 
 CFLAGS	?= -O2
-WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+WARNING_CFLAGS ?= -Wall -Wextra -Werror -Wdeclaration-after-statement
```

Comment:
I quite like this idea. I also think it is good to catch warnings early. However, I think that we should do this with CMake as well and not just Makefiles...

Reply:
mmm, I do not think we do. Just to check I ran the following commands:
```
mkdir build && cd build
cmake ..
make VERBOSE=1
```
And I got the following output:
```
[  0%] Building C object library/CMakeFiles/mbedcrypto.dir/aes.c.o
cd /Users/andama01/Repos/mbedtls-cpy/build/library && /Library/Developer/CommandLineTools/usr/bin/cc  -I/Users/andama01/Repos/mbedtls-cpy/include  -Wall -Wextra -W -Wdeclaration-after-statement -Wwrite-strings -Wpointer-arith -Wimplicit-fallthrough -Wshadow -Wmissing-declarations -Wmissing-prototypes -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk   -o CMakeFiles/mbedcrypto.dir/aes.c.o   -c /Users/andama01/Repos/mbedtls-cpy/library/aes.c
[  0%] Building C object library/CMakeFiles/mbedcrypto.dir/aesni.c.o
cd /Users/andama01/Repos/mbedtls-cpy/build/library && /Library/Developer/CommandLineTools/usr/bin/cc  -I/Users/andama01/Repos/mbedtls-cpy/include  -Wall -Wextra -W -Wdeclaration-after-statement -Wwrite-strings -Wpointer-arith -Wimplicit-fallthrough -Wshadow -Wmissing-declarations -Wmissing-prototypes -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk   -o CMakeFiles/mbedcrypto.dir/aesni.c.o   -c /Users/andama01/Repos/mbedtls-cpy/library/aesni.c
```
So I do not think that CMake uses -Werror (at least not for Clang anyways), see [here](https://github.com/ARMmbed/mbedtls/blob/development/CMakeLists.txt#L138)